### PR TITLE
Dscanner 0.3.0 (new formula)

### PR DESCRIPTION
**Pull request by SommerEngineering**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-core/blob/master/.github/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally prior to submission with `brew install <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [x] Does your submission pass `brew audit --new-formula <formula>` (after doing `brew install <formula>`)?

---

Added the Dscanner utility for the D programming language

### CODE REVIEW

File: Formula/dscanner.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
+class Dscanner < Formula
+  desc "The Dscanner untility for the D programming language"
```
Comment (bfontaine) [inline Formula/dscanner.rb]: This doesn’t really tell us what this utility is for. Could you please write a more descriptive description?

File: Formula/dscanner.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  test do
+    (testpath/"test.d").write <<-EOS.undent
+    import std.stdio;
+    void main(string[] args)
+    {
+       writeln("Hello World");
+    }
+    EOS
+
+    #system "#{bin}/dscanner", "--tokenCount", "test.d"
```
Comment (bfontaine) [inline Formula/dscanner.rb]: Please remove this comment.

File: Formula/dscanner.rb [hunk 3]
```diff
@@ -0,0 +1,25 @@
+class Dscanner < Formula
+  desc "A tool to analyse e.g. the style and syntax of D code."
+  homepage "https://github.com/Hackerpilot/Dscanner"
+  url "https://github.com/Hackerpilot/Dscanner/archive/v0.3.0.tar.gz"
+  sha256 "2b5578ca98ad6805a1f1494dfbf90c8f491da20bfe5103ca0c8ac73d781558fe"
+  depends_on "dmd" => :build
```
Comment (MikeMcQuaid) [inline Formula/dscanner.rb]: Newline above this line.

File: Formula/dscanner.rb [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+  depends_on "dub" => :build
+
+  def install
+    system "dub", "build"
+    bin.install "dscanner"
```
Comment (MikeMcQuaid) [inline Formula/dscanner.rb]: Is there no `dub install` that can be used or documentation to be installed?

File: Formula/dscanner.rb [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+  end
+
+  test do
+    (testpath/"test.d").write <<-EOS.undent
+    import std.stdio;
```
Comment (MikeMcQuaid) [inline Formula/dscanner.rb]: Indent these lines by two chars.

**Comment by SommerEngineering** [comment]
@bfontaine, @MikeMcQuaid: I have done all changes and all tests are passed. @MikeMcQuaid: There is no `dub install` for D, as in Go.

**Comment by MikeMcQuaid** [comment]
Thanks for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

**Pull request closed without merging**